### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by craigs40**
In this project, I've built an application that allows users to create posts and the authors of the posts are displayed only if you are logged in.

### CODE REVIEW

**Review by MkrtichSargsyan — COMMENTED** [review]
## _Status:_ Invalid submission warning

## Hello Shay Craig


This Code Review Request will be marked as invalid in your Dashboard because you didn't follow the correct [Git Flow](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59882084/) as required for this project. 

- [ ] Kindly create  ```main``` branch

Please make the necessary changes and submit another code review request once you are ready! :slightly_smiling_face:.


> ps: . If you have any questions about my review or need help, you can message me on slack ( @mkrtichsargsyan )

**Review by BrianSammit — CHANGES_REQUESTED** [review]
CHANGES REQUIRED :red_circle: :stop_sign: 
---------------
Hello @craigs40, great job you are doing on this project :clap: :muscle: 
TO HIGHLIGHT: 

However there are some changes to make to complete this milestone :hammer_and_wrench: see them below :point_down: :eyes: 
---------

- [ ] Create the associations between users and posts.

associations_hint_requirement

- [ ] I cannot create a new post. Instead the page is presented with a form for a new user. 
Screenshot from 2021-06-08 11-10-54

- [ ] Create the route with the ` only: [:new, :create, :index]` 

posts_routes_requirement (1)

- [ ] As I ask you in the first requirement in the post model, create the association `belongs_to a user`
- [ ] Please check the step on Getting Started on your readme file. Before you run `rails server` you need to run `rails db:create` and `rails db:migrate`. Please add these commands
- [ ] [OPTIONAL] Highlight the command that they need to run example `commands`. Just put the commands inside backticks ` `` `
---------------
Please make these require changes and send it again for another review :+1: :muscle: 
Reach me on SLACK if you have any questions. 
@ Brian Sammit Cruz Rodriguez
HAPPY CODING!!!

MB5W4D

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class Post < ApplicationRecord
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (cyonii) [inline app/models/post.rb]: The devise gem is used to configure an authentication system for your user model, please remove this code here as it appears to be useless.

**Review by cyonii — CHANGES_REQUESTED** [review]
### STATUS: CHANGES REQUIRED ♻️✍️

Hello team, keep up the energy.

#### TO IMPROVE (MANDATORY)
You have done great so far, however, these few defects are holding your project back from getting approved, kidnly fix them and resubmit.
- [ ] Your app throws an error with the message "**unknown attribute 'title' for Post.**" when I try to add a new post. Kindly fix this.
> Kindly note that the next reviewer may ask for changes that has not been stated here due to this error not allowing the rest of the project be reviewed.

- [ ] As the previous reviewer pointed out, please add the necessary commands needed to set up this project on your README. For example, I needed to run `bundle install` but you failed to mention that.


> #### <br>You can ping me on Slack @[CY Kalu](https://microverse-students.slack.com/team/U01494U96FR) to draw my attention in case I do not respond quickly on here. I'm always available and will be glad to help.<br><br>

**Review by Georjane — CHANGES_REQUESTED** [review]
## Required Changes ♻️
### Hello, Team <STUDENTS>, ✋🏽 

Great work on making the changes requested by the previous reviewer 👏 
You've done well implementing some of the requested changes, however, you seem to have introduced a bug during the changes you were making.

- [ ] Right now I am unable to signup for your application to test any of its functionalities. I get the following error:

image
Please check this and submit for another code review

Cheers and Happy coding! 💻 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 🤝 

_Code Reviewer
**[Jane](https://witahgeorjane.tech/)**_

**Review by cyonii — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️✍️

You have shown impressive resilience so far, I want personally to encourage you to keep your heads up.

### TO IMPROVE (MANDATORY)
- [ ] Exactly the same issue the previous reviewer reported still exists, therefore I am unable to conduct a review. Please fix this issue and resubmit. Review `devise`'s [documentation](https://github.com/heartcombo/devise) and make sure you're doing everything tight
Screenshot 2021-07-02 at 8 47 07 PM


> #### <br>You can ping me on Slack @[CY Kalu](https://microverse-students.slack.com/team/U01494U96FR) to draw my attention in case I do not respond quickly on here. I'm always available and will be glad to help.<br><br>

**Comment by craigs40** [comment]
@Georjane @cyonii unfortunately, i do not see this error message when i run this rails app and, therefore, im unsure how to proceed

**Review by cyonii — APPROVED** [review]
# STATUS: APPROVED ✅

Thank you @craigs40 for implementing the required changes. Your project has now met the requirements and has been approved. Congratulations 🎉.

You can now merge this PR and move on. Happy coding 🤓💻

> #### <br>You can ping me on Slack @[CY Kalu](https://microverse-students.slack.com/team/U01494U96FR) to draw my attention in case I do not respond quickly on here. I'm always available and will be glad to help.<br><br>

**Pull request merged**